lib/lazy_object.rb

```diff
@@ -27,4 +27,17 @@ def __target_object__
   def method_missing(method_name, *args, &block)
     __target_object__.send(method_name, *args, &block)
   end
+
+  def present?
+    !blank?
+  end
+
+  def blank?
+    __target_object__.respond_to?(:empty?) ? !!__target_object__.empty? : !__target_object__
+  end
+
+  # Forwards method calls to the target object.
```

Comment:
@skarm61 since you are inheriting from BasicObject, not sure how necessary the rest of your changes are. Come to think of it, please write some unit tests for this.